Fix Snowflake hook conn id

dhuang opened the pull request:
Do not set manually to ensure the base `DbApiHook` correctly interprets
arguments based on `conn_name_attr` and `default_conn_name`.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

ashb commented on tests/providers/snowflake/hooks/test_snowflake.py:
```diff
@@ -102,6 +102,7 @@ def test_get_conn_params(self):
                                 'warehouse': 'af_wh',
                                 'region': 'af_region',
                                 'role': 'af_role'}
+        self.assertEqual(self.db_hook.snowflake_conn_id, 'snowflake_default')
```
I _think_ that this is actually a change in the default, and `snowflake_conn_id` was the previous default.

Do you think it's worth changing this default? If so it needs a note in UPDATING.md

dhuang replied:
Hmm yeah that could've been the intent. However, with the way it was implemented it, I believe it also would have overwrote an explicit `snowflake_conn_id` passed into the hook/operator on initialization. But good call, I think it does make sense to note this in `UPDATING.md`, added.

dhuang committed:
Fix Snowflake hook conn id

Do not set manually to ensure the base `DbApiHook` correctly interprets
arguments based on `conn_name_attr` and `default_conn_name`.
